### PR TITLE
vc pull request

### PR DESCRIPTION
**Pull request by xiaochunxin**

### CODE REVIEW

File: athena/data/datasets/voice_conversion.py [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    def load_csv(self, file_path):
+        """ load csv file """
+        return self.preprocess_data(file_path)
+
+    def reload_config(self, config):
+        """ reload the config """
+        if config is not None:
+            self.hparams.override_from_dict(config)
+
+    def pad_wav_to_get_fixed_frames(self, x: np.ndarray, y: np.ndarray):
```
Comment (Some-random) [inline athena/data/datasets/voice_conversion.py]: remove this function if its not needed

File: athena/data/datasets/voice_conversion.py [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+            log_f0s_concatenated = np.ma.log(np.concatenate(f0s))
+            log_f0s_mean = log_f0s_concatenated.mean()
+            log_f0s_std = log_f0s_concatenated.std()
+            cmvns.append(
+                (speaker, coded_sps_mean, coded_sps_std, log_f0s_mean, log_f0s_std))
+
+        df = pandas.DataFrame(data=cmvns, columns=["speaker", "codedsp_mean", "codedsp_var", "f0_mean", "f0_var"])
+        df.to_csv(self.hparams.cmvn_file, index=False, sep="\t")
+
+        # feature_dim = self.audio_featurizer.dim * self.audio_featurizer.num_channels
```
Comment (Some-random) [inline athena/data/datasets/voice_conversion.py]: remove these comments

File: athena/loss.py [hunk 3]
```diff
@@ -97,3 +97,77 @@ def __call__(self, logits, samples, logit_length=None):
         loss = tf.reduce_sum(tf.abs(loss, name="L1_loss"), axis=-1)
         loss = tf.reduce_mean(loss)
         return loss
+
+def GeneratorLoss(discirmination, input_real, generated_back,identity_map,\
+                                           target_label_reshaped, domain_out_real):
+    lambda_cycle = 10
```
Comment (Some-random) [inline athena/loss.py]: make these hyperparams or at least write some explanation for these magic number

File: athena/loss.py [hunk 4]
```diff
@@ -97,3 +97,77 @@ def __call__(self, logits, samples, logit_length=None):
...
+    domain_real_loss = tf.reduce_mean(
+        tf.nn.softmax_cross_entropy_with_logits(labels=target_label_reshaped, logits=domain_out_real))
+    return domain_real_loss
+
+class StarganLoss(tf.keras.losses.Loss):
+    def __init__(self, name="StarganLoss"):
+        super().__init__(name=name)
+    def __call__(self, outputs,samples, logit_length=None):
+        input_real, target_real, target_label, source_label = samples["src_coded_sp"], \
+                                                              samples["tar_coded_sp"], samples["tar_speaker"], \
```
Comment (Some-random) [inline athena/loss.py]: this line is still super long?

**Comment by tjadamlee** [comment]
please restart pr after you have finish your implement
